tests/test_node_backup_role.py

```
@@ -0,0 +1,319 @@
+    failing — a source that looks configured and backs up nothing.
+    """
+    ctx: dict[str, object] = dict(_defaults())
+    ctx.update(
+        ansible_managed="Ansible managed",
+        inventory_hostname="beelink",
+        node_backup_sources=BEELINK_SOURCES,
+        node_backup_restic_version="0.19.1",
+        node_backup_r2_repo_prefix="s3:https://acct.r2.cloudflarestorage.com/kubelab-backups",
+    )
+    ctx.update(overrides)
+    env = Environment(
```

Ruff (0.16.1)

[error] By default, jinja2 sets `autoescape` to `False`. Consider using `autoescape=True` or the `select_autoescape` function to mitigate XSS vulnerabilities.

(S701)